molecule-runtime/src/quartzCoreMain/kotlin/app/cash/molecule/DisplayLinkClock.kt

```diff
@@ -0,0 +1,33 @@
+package app.cash.molecule
+
+import androidx.compose.runtime.BroadcastFrameClock
+import androidx.compose.runtime.MonotonicFrameClock
+import kotlinx.cinterop.ObjCAction
+import platform.Foundation.NSRunLoop
+import platform.Foundation.NSSelectorFromString
+import platform.QuartzCore.CADisplayLink
+
+public actual object DisplayLinkClock : MonotonicFrameClock {
+
+  @Suppress("unused") // This registers a DisplayLink listener.
+  private val displayLink: CADisplayLink = CADisplayLink.displayLinkWithTarget(
+    target = this,
+    selector = NSSelectorFromString(this::tickClock.name),
+  )
+
+  private val clock = BroadcastFrameClock {
+    // We only want to listen to the DisplayLink run loop if we have frame awaiters.
+    displayLink.addToRunLoop(NSRunLoop.currentRunLoop, NSRunLoop.currentRunLoop.currentMode)
+  }
+
+  override suspend fun <R> withFrameNanos(onFrame: (frameTimeNanos: Long) -> R): R {
+    return clock.withFrameNanos(onFrame)
+  }
+
+  @ObjCAction public fun tickClock() {
```

Comment:
Apparently functions used by `NSSelectorFromString` need to be public or else they don't work. I haven't been able to confirm this because unit tests don't currently work.

Reply:
This is probably better served as an inline comment than a github PR comment.

Reply:
Makes sense since it's basically being used a callback from outside the class. With `@ObjCName` in Kotlin 1.8 we may be able to give it a stable name and have it still work when private, but I'm not sure I care all that much.